docs/architecture/app-class-and-template.md

```diff
@@ -0,0 +1,200 @@
+# ApplicationClass and ApplicationTemplate
+
+Managing Applications can be passed on some external system, to remove the burden of integration with Compass from Applications.
+Because of that, we are going to extend our API with IntegrationSystem and ApplicationClass types.
+
+To increase the reusability of the application configuration, we introduce ApplicationTemplate. 
+
+
+## Terminology
+
+- **Application** represents functionality that provides API or Events.
+
+- **ApplicationClass** represents soon-to-be Application that is manged by IntegrationSystem. 
+   
+- **ApplicationTemplate** represents soon-to-be Application with all details already provided (like API and Events) that 
+can be created by providing values for replacing placeholders.
+  
+- **IntegrationSystem** (abbreviation: **IntSys**):
+
+    Any system that works in the context of multiple tenants, managing multiple Applications or Runtimes. It exists in a global scope without a tenant assignment.
+
+## GraphQL API Changes
+
+1. Managing Integration Systems
+ 
+```
+type IntegrationSystem {
+    id: ID
+    name: String
+    description: String
+    auths: [SystemAuth]
+}
+
+input IntegrationSystemInput {
+    name: String!
+    description: String
+}
+
+type Mutation {
+    createIntegrationSystem(in: IntegrationSystemInput!): IntegrationSystem!
+    updateIntegrationSystem(id: ID!, in: IntegrationSystemInput!): IntegrationSystem!
+    deleteIntegrationSystem(id: ID!): IntegrationSystem! 
+    
+    addCredentialsForIntegrationSystem(id: ID!): SystemAuth!
+    removeCredentialsForIntegrationSystem(id: ID!): SystemAuth!
+}
+
+type Query {
+    integrationSystems(first: Int = 100, after: PageCursor): IntegrationSystemPage!
+    integrationSystem(id: ID!): IntegrationSystem
+}
+```
+Removing Integration System can be performed only when:
+- there is no Application Class from this Integration System
+- there is no Application managed by the Integration System 
+
+   
+2. Managing Application Class
+
+```
+input ApplicationClassInput {
```

Comment:
Do we assume that on this level (ApplicationClass) the API specs are defined? Not sure if this is still valid, but there was a scenario when a particular instance of the Application can provide only a subset of its API for given spec credentials.

Reply:
In the first implementation, we make an assumption that ApplicationClass will be registered globally. Because of that, every tenant will see the same API. 
API is defined in `apiSpecs` and `eventAPISpecs` and they are just for documentation purposes. AS a user I would like to know what kind of API given ApplicationClass can provide. Notice that this is APISpecINput, no APIDefinitionInput so no credentials are defined at this level.